tests/testutils.py

```diff
@@ -19,14 +19,50 @@
 middleware = ComponentDependencyMiddleware(get_response=lambda _: response_stash)
 
 
+class GenIdPatcher:
```

Comment:
As a reminder, internally we use a "ID generator" which returns unique IDs, so that each component, template tag node, and even each time we render a component all have their unique IDs. But in tests this is problematic, because it's not deterministic. So we patch the ID generator, to make deterministic.

I moved this logic for the patching of the random ID generator out, so I could reuse it.

And the benchmarked code also used the CSRF token, so I also added the `CsrfTokenPatcher` below, so that the CSRF token returns `predictabletoken` in the tests.